mappers/bluetooth_mapper/action_manager/action_manager.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2019 The KubeEdge Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package action_manager
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/paypal/gatt"
+)
+
+// GattPeripheral  represents the remote gatt peripheral device
+var GattPeripheral gatt.Peripheral
+
+// Operation is structure to define device operation
+type Operation struct {
+	// Action can be one of read/write corresponding to get/set respectively
+	Action string `yaml:"action" json:"action"`
+	// Characteristic is  refers to the characteristic on which the operation needs to be performed
+	CharacteristicUUID string `yaml:"characteristic-uuid" json:"characteristic-uuid"`
+	// Value is the value to be written in case of write action and value read from the device in case of read action
+	Value []byte `yaml:"value" json:"value"`
+}
+
+// Action is structure to define a device action
+type Action struct {
+	Enable bool `yaml:"enable" json:"enable"`
+	// Name is the name of the Action
+	Name string `yaml:"name" json:"name"`
+	// Operation specifies the operation to be performed for this action
+	Operation Operation `yaml:"operation" json:"operation"`
+}
+
+//ActionManager is a structure that contains a list of actions
+type ActionManager struct {
+	Actions []Action `yaml:"actions"`
+}
+
+//PerformOperation executes the operation
+func (action *Action) PerformOperation() {
+	glog.Infof("Performing operations associated with action:  %s", action.Name)
+	characteristic, err := FindCharacteristic(GattPeripheral, action.Operation.CharacteristicUUID)
+	if err != nil {
+		glog.Errorf("Error in finding characteristics: %s", err)
+	}
+	if strings.ToUpper(action.Operation.Action) == "READ" {
+		readValue, err := ReadCharacteristic(GattPeripheral, characteristic)
+		if err != nil {
+			glog.Errorf("Error in reading  characteristic: %s", err)
+			return
+		}
+		//todo -- Pass readValue to data-converter
+		action.Operation.Value = readValue
+		glog.Infof("Read Successful")
```

Comment:
Infof => Info

Reply:
I think Infof is fine.

Reply:
ACK. I tested by go fmt & go vet, It is ok

Reply:
@edisonxiang does this apply for the other comments as well or do u want me to change in the other places that have been mentioned ?

Reply:
@sujithsimon22 It is also fine for the other places:)

Reply:
great, thanks :)